File: backend/src/application/commands/delete_library_item.py
from __future__ import annotations

from dataclasses import dataclass

from domain.repositories import LibraryRepository


@dataclass(frozen=True)
class DeleteLibraryItemCommand:
    item_id: str


class DeleteLibraryItemHandler:
    def __init__(self, repository: LibraryRepository) -> None:
        self._repository = repository

    def handle(self, command: DeleteLibraryItemCommand) -> None:
        self._repository.delete(command.item_id)

File: frontend/src/application/commands/DeleteLibraryItem.ts
import { LibraryRepository } from '@domain/repositories/LibraryRepository';

export interface DeleteLibraryItemCommand {
  itemId: string;
}

export class DeleteLibraryItem {
  constructor(private repository: LibraryRepository) {}

  handle(command: DeleteLibraryItemCommand): Promise<void> {
    return this.repository.delete(command.itemId);
  }
}
